api/apps/v1alpha1/nemo_customizer_types.go

```diff
@@ -0,0 +1,961 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"maps"
+	"os"
+	"strconv"
+
+	rendertypes "github.com/NVIDIA/k8s-nim-operator/internal/render/types"
+	utils "github.com/NVIDIA/k8s-nim-operator/internal/utils"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	// NemoCustomizerConditionReady indicates that the NEMO CustomizerService is ready.
+	NemoCustomizerConditionReady = "Ready"
+	// NemoCustomizerConditionFailed indicates that the NEMO CustomizerService has failed.
+	NemoCustomizerConditionFailed = "Failed"
+
+	// NemoCustomizerStatusPending indicates that NEMO CustomizerService is in pending state
+	NemoCustomizerStatusPending = "Pending"
+	// NemoCustomizerStatusNotReady indicates that NEMO CustomizerService is not ready
+	NemoCustomizerStatusNotReady = "NotReady"
+	// NemoCustomizerStatusReady indicates that NEMO CustomizerService is ready
+	NemoCustomizerStatusReady = "Ready"
+	// NemoCustomizerStatusFailed indicates that NEMO CustomizerService has failed
+	NemoCustomizerStatusFailed = "Failed"
```

Comment:
I see all NeMo microservices use similar status type.
Maybe we should consider moving them under `common_types.go`

Reply:
yes, makes sense, can do that as a follow up.